improver/cli/subset_refrun_data.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2020 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to extract a spatial subset of input file data."""
+
+from improver import cli
+from improver.cli import parameters
+
+
+@cli.clizefy
+@cli.with_output
+def process(
```

Comment:
This is a good question. I think in this case the need is better served by having a separate CLI.  This is for two main reasons:

1. The implementation of the reference run suite doesn't have an easy way to distinguish between spot and gridded data.  So ideally we would have a single interface to both grid and site list subsetting.
2. The extract functionality is based heavily around the expectation of equality constraints, rather than range constraints.  This assumption is hard-coded throughout the functions in the `cube_extraction` module.  I think this would require quite significant refactoring or a new function to be able to support subsetting of a grid, as opposed to extracting a single threshold (or list of thresholds), time slice, height or pressure level (which is what the extract CLI was designed to do).

Given then that we would need a new function to do the grid subsetting, and that it introduces complexity into the suite to have to distinguish between spot and gridded data, I think it is worth the cost of a small amount of duplicated code for site extraction to have this as a separate CLI.